server/src/main/java/org/opensearch/cluster/metadata/IndexMetadata.java

```diff
@@ -1278,7 +1278,7 @@ public void writeVerifiableTo(BufferedChecksumStreamOutput out) throws IOExcepti
         out.writeByte(state.id());
         writeSettingsToStream(settings, out);
         out.writeVLongArray(primaryTerms);
-        out.writeMapValues(mappings, (stream, val) -> val.writeTo(stream));
+        out.writeMapValues(mappings, (stream, val) -> val.writeVerifiableTo((BufferedChecksumStreamOutput) stream));
```

Comment:
Isn't `BufferedChecksumStreamOutput` inheriting `StreamOutput`?

Reply:
Yes, but in this case the default `writeMapValues` has `StreamOutput` interface and the lambda function assumes that, so we need to cast to `BufferedChecksumStreamOutput`